pkg/errors/errors.go

```diff
@@ -0,0 +1,153 @@
+package errors
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Kind enums
+const (
+	KindNotFound   = http.StatusNotFound
+	KindBadRequest = http.StatusBadRequest
+	KindUnexpected = http.StatusInternalServerError
+)
+
+// Error is an Athens system error.
+// It carries information and behavior
+// as to what caused this error so that
+// callers can implement logic around it.
+type Error struct {
+	// Kind categories Athens errors into a smaller
+	// subset of errors. This way we can generalize
+	// what an error really is: such as "not found",
+	// "bad request", etc. The official categories
+	// are HTTP status code but the ones we use are
+	// imported into this package.
+	Kind     int
+	Op       Op
+	Module   M
+	Version  V
+	Err      error
+	Severity logrus.Level
+}
+
+// Error returns the underlying error's
+// string message. The logger takes care
+// of filling out the stack levels and
+// extra information.
+func (e Error) Error() string {
+	return e.Err.Error()
+}
+
+// Op describes any independent function or
+// method in Athens. A series of operations
+// forms a more readable stack trace.
+type Op string
+
+// M represents a module in an error
+// this is so that we can distinguish
+// a module from a regular error string or version.
+type M string
+
+// V represents a module version in an error
+type V string
+
+// E is a helper function to construct an Error type
+// Operation always comes first, module path and version
+// come second, they are optional. Args must have at least
+// an error or a string to describe what exactly went wrong.
+// You can optionally pass a Logrus severity to indicate
+// the log level of an error based on the context it was constructed in.
+func E(op Op, args ...interface{}) error {
+	if len(args) == 0 {
+		panic("errors.E called with 0 args")
```

Comment:
I'm not sure of a better option but I prefer to avoid calling panic anywhere other than main.

Reply:
Maybe similar to line 85 below, we should return an error with a message reporting that no arguments were passed.

Reply:
will do